sds_data_manager/lambda_code/SDSCode/pipeline_lambdas/spice_indexer.py

```diff
@@ -371,87 +371,72 @@ def index_pointing_data(s3_key: str):
-                    pointing_end_utc = pd.to_datetime(
-                        next_row.iloc[0]["repoint_end_utc"]
-                    )
-                    repoint_start_utc = pd.to_datetime(
-                        next_row.iloc[0]["repoint_start_utc"]
-                    )
-                    repoint_end_utc = pd.to_datetime(
-                        next_row.iloc[0]["repoint_end_utc"]
-                    )
-                else:
-                    repoint_start_utc = None
-                    repoint_end_utc = None
-                    pointing_end_utc = None
-
-            except Exception as e:
-                logger.error(
-                    f"Error calculating pointing UTCs for repoint_id {repoint_id}: {e}"
-                )
-                continue
-
-            params = {
+    max_repoint_id = repoint_df["repoint_id"].max()
+    repoint_records = []
+
+    for repoint_id in repoint_df["repoint_id"].values:
+        # Had to convert to match the type in the database
+        repoint_id = int(repoint_id)  # noqa: PLW2901
+        # First find if next row exists
+        if repoint_id < max_repoint_id:
+            # If next row exists, calculate the pointing data
+            row_data = {
                 "pointing_id": repoint_id,
-                "pointing_start_utc": pointing_start_utc,
-                "pointing_end_utc": pointing_end_utc,
-                "repoint_start_utc": repoint_start_utc,
-                "repoint_end_utc": repoint_end_utc,
+                "pointing_start_utc": datetime.strptime(
+                    repoint_df.loc[
+                        repoint_df["repoint_id"] == repoint_id, "repoint_end_utc"
+                    ].values[0],
+                    "%Y-%m-%dT%H:%M:%S.%f",
+                ),
+                "pointing_end_utc": datetime.strptime(
+                    repoint_df.loc[
+                        repoint_df["repoint_id"] == repoint_id + 1, "repoint_end_utc"
+                    ].values[0],
+                    "%Y-%m-%dT%H:%M:%S.%f",
+                ),
+                "repoint_start_utc": datetime.strptime(
+                    repoint_df.loc[
+                        repoint_df["repoint_id"] == repoint_id + 1, "repoint_start_utc"
+                    ].values[0],
+                    "%Y-%m-%dT%H:%M:%S.%f",
+                ),
+                "repoint_end_utc": datetime.strptime(
+                    repoint_df.loc[
+                        repoint_df["repoint_id"] == repoint_id + 1, "repoint_end_utc"
+                    ].values[0],
+                    "%Y-%m-%dT%H:%M:%S.%f",
+                ),
             }
+        else:
+            row_data = {
+                "pointing_id": repoint_id,
+                "pointing_start_utc": datetime.strptime(
+                    repoint_df.loc[
+                        repoint_df["repoint_id"] == repoint_id, "repoint_end_utc"
+                    ].values[0],
+                    "%Y-%m-%dT%H:%M:%S.%f",
+                ),
```

Comment:
Move this outside of the loop and do it at the end.

```suggestion
            row_data = {
                "pointing_id": repoint_id,
                "pointing_start_utc": datetime.strptime(
                    repoint_df.iloc[-1]["repoint_end_utc"].values[0],
                    "%Y-%m-%dT%H:%M:%S.%f",
                ),
```